File: android/app/src/main/java/com/lightningkite/kwifttemplate/shared/views/WeekDemoVG.kt
package com.lightningkite.kwifttemplate.shared.views

import android.view.View
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwift.views.shared.ViewGenerator
import com.lightningkite.kwifttemplate.layouts.WeekDemoXml

class WeekDemoVG() : ViewGenerator() {
    override val title: String get() = "Week Demo"

    override fun generate(dependency: ViewDependency): View {
        val xml = WeekDemoXml()
        val view = xml.setup(dependency)

        return view
    }
}

File: ios/Kwift Template/com/lightningkite/kwifttemplate/shared/views/WeekDemoVG.swift
//Package: com.lightningkite.kwifttemplate.shared.views
//Converted using Kwift2

import Foundation
import Kwift



public class WeekDemoVG: ViewGenerator {
    
    
    override public var title: String {
        get {
            return "Week Demo"
        }
    }
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = WeekDemoXml()
        var view = xml.setup(dependency)
        xml.week.bind(data: ConstantObservableProperty{ (start, end) in 
            [WeekViewEvent(id: 0, title: "Event", start: Date(), end: Date() + 1.hours(), colorRes: ResourcesColors.colorPrimary)]
        }, onEventClick: { (it) in 
            print(it)
        }, onEmptyClick: { (it) in 
            print(it)
        })
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    override public init() {
        super.init()
    }
}
 
